packages/snap/src/polkadot/events/index.ts

```diff
@@ -53,6 +54,13 @@ class EventEmitterPolkadotImplementation implements EventEmitter<PolkadotEvent>
     return this;
   }
 
+  numberOfListeners(event: PolkadotEvent, origin: string): number {
+    if (this.hasAttachedListeners(event, origin)) {
+      return this.listeners[origin][event].length;
```

Comment:
It will explode if there is origin but not event

Reply:
Actually that is checked inside `this.hasAttachedListeners`